.github/workflows/versions.yml

```
@@ -168,6 +168,21 @@ jobs:
       - name: Verify node
         run: __tests__/verify-node.sh 20
 
+  version-file-dev-engines:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest, macos-13]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup node from node version file
+        uses: ./
+        with:
+          node-version-file: '__tests__/data/package-dev-engines.json'
+      - name: Verify node
+        run: __tests__/verify-node.sh 20
+
```

```suggestion
  version-file-dev-engines-array:
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest, macos-13]
    steps:
      - uses: actions/checkout@v4
      - name: Setup node from node version file
        uses: ./
        with:
          node-version-file: '__tests__/data/package-dev-engines-array.json'
      - name: Verify node
        run: __tests__/verify-node.sh 20

```

suggested file content:

```json
{
  "engines": {
    "node": "^19"
  },
  "devEngines": {
    "runtime": [
      {
        "name": "bun",
        "version": "^1"
      },
      {
        "name": "node",
        "version": "^20"
      }
    ]
  }
}
```